app/main.py

```diff
@@ -1,2 +1,80 @@
+from typing import Hashable, Any
+
+
 class Dictionary:
-    pass
+    def __init__(self) -> None:
+        self.capacity = 8
+        self.table = [[] for _ in range(self.capacity)]
+        self.size = 0
+        self.load_factor = 0.67
+
+    def __len__(self) -> int:
+        return self.size
+
+    def __setitem__(self, key: Hashable, value: Any) -> None:
+        if self.size >= len(self.table) * self.load_factor:
+            self.resize()
+        index = hash(key) % len(self.table)
+        bucket = self.table[index]
+        for i, (k, v) in enumerate(bucket):
+            if k == key:
+                bucket[i] = (key, value)
+                break
+        else:
+            bucket.append((key, value))
+            self.size += 1
```

Comment:
what about collisions ?

Reply:
my variant works great with collisions, although not like others
![image](https://user-images.githubusercontent.com/46852739/221906644-b47e7ad2-566f-495e-b5b3-8292efef9d38.png)

Reply:
my 'bucket' is a mutable list, with 2 elements tuples.
I keep collisions there and have fast and simple access to them.
it is another way implement all requirements of the task

Reply:
do you know what collision is?) And where is the collision on your screenshot?)

Reply:
it is a good way to implement the task, u are saving a few nodes in one bucket in case of collisions, but it is not how dict work in python ((
![image](https://user-images.githubusercontent.com/103201010/223114528-f48d82f5-a10d-4d8b-b891-1fb5b6e2c42d.png)